Documentation/ApiOverview/UpdateWizards/Creation.rst

```diff
@@ -7,69 +7,114 @@ Creating generic update wizards
-           return true;
-       }
+		/**
+		 * Return the identifier for this wizard
+		 * This should be the same string as used in the ext_localconf class registration
+		 *
+		 * @return string
+		 */
+		public function getIdentifier(): string
+		{
+			return 'exampleUpdateWizard';
+		}
+
+		/**
+		 * Return the speaking name of this wizard
+		 *
+		 * @return string
+		 */
+		public function getTitle(): string
+		{
+			return 'Title of this updater';
+		}
+
+		/**
+		 * Return the description for this wizard
+		 *
+		 * @return string
+		 */
+		public function getDescription(): string
+		{
+			return 'Description of this updater';
+		}
+
+		/**
+		 * Execute the update
+		 *
+		 * Called when a wizard reports that an update is necessary
+		 *
+		 * @return bool
+		 */
+		public function executeUpdate(): bool
+		{
+
+		}
+
+		/**
+		 * Is an update necessary?
+		 *
+		 * Is used to determine whether a wizard needs to be run.
+		 * Check if data for migration exists.
+		 *
+		 * @return bool
+		 */
+		public function updateNecessary(): bool
+		{
+
+		}
+
+		/**
+		 * Returns an array of class names of Prerequisite classes
+		 *
+		 * This way a wizard can define dependencies like "database up-to-date" or
+		 * "reference index updated"
+		 *
+		 * @return string[]
+		 */
+		public function getPrerequisites(): array
+		{
+
+		}
    }
 
-Property :php:`$title`
-   Can be overwritten to define the title used while rendering the list of available
-   update wizards.
+Method :php:`getIdentifier`
+   Return the identifier for this wizard. This should be the same string as used 
+   in the ext_localconf class registration.
 
-Method :php:`checkForUpdate`
-   Is called to check whether the updater has to run. Therefore a boolean has to be
-   returned. The :php:`$description` provided can be modified as a reference to
-   provide further explanation, in addition to the title.
+Method :php:`getTitle`
+   Return the speaking name of this wizard.
+
+Method :php:`getDescription`
+   Return the description for this wizard.
 
-Method :php:`performUpdate`
+Method :php:`executeUpdate`
    Is called if the user triggers the wizard. This method should contain, or call,
-   the code that is needed to execute the update. :php:`$databaseQueries` and
-   :php:`$customMessage` can be used as reference to provide further information to
-   the user after the update function is completed. :php:`$databaseQueries` has to
-   be an array, where each value is a string containing the query. This array should
-   only contain executed queries. :php:`$customMessage` is a string, where further
-   information is provided for the user after the updated process has completed.
+   the code that is needed to execute the update. 
+
+Method :php:`updateNecessary`
+   Is called to check whether the updater has to run. Therefore a boolean has to be
+   returned.
+   
+Method :php:`getPrerequisites`
+   Returns an array of class names of Prerequisite classes. This way a wizard can 
```

Comment:
Why uppercase `P`? Any maybe add some specific example here or in the code above.

Reply:
Copy paste from functions header ;-) Adding an example is a good idea. Thanks!